### PR TITLE
docs: rewrite private npm module support

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

Basically a full rewrite.

Major changes:

- Use ordered lists where possible
- Use present tense and active voice
- Grammar fixes
- Use monospaced font for some filenames
- Rename filename so that it is npm specific

I've tried to clarify some difficult to read text, but this needs a review from the maintainers to verify if the text is correct.

## Context:

Helps with #7626.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/private-modules.md [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+Assuming the private module lookup succeeds then private package versions will be kept up-to-date like the public package versions are.
+
+### Lock file generation
+
+If you use a lock file (e.g. Yarn's `yarn.lock` or npm's `package-lock.json`) then Renovate must update that lock file whenever _any_ package listed in your package file is updated to a new version.
+
+To do this, Renovate runs `npm install` or equivalent and saves the resulting lock file.
 If a private module hasn't been updated, it _usually_ won't matter to npm/Yarn because they won't attempt to update its lock file entry anyway.
 However it's possible that the install will fail if it attempts to look up that private module for some reason, even when that private module is not the main one being updated.
 It's therefore better to provide Renovate with all the credentials it needs to look up private packages.
```
Comment (HonkingGoose) [inline docs/usage/private-modules.md:37]: This is really hard to understand for me.
I would rewrite this myself but I don't really understand the text we have now, so I'll leave this for the maintainers to fix. 😄

File: docs/usage/private-modules.md [hunk 2]
```diff
@@ -59,18 +65,19 @@ module.exports = {
 
 ### Commit .npmrc file into repository
 
-One approach that many projects use for private repositories is to simply check in an authenticated `.npmrc` into the repository that is then shared between all developers.
+One approach that many projects use for private repositories is to simply check-in an authenticated `.npmrc` into the repository that is then shared between all developers.
 Therefore anyone running `npm install` or `yarn install` from the project root will be automatically authenticated with npm without having to distribute npm logins to every developer and make sure they've run `npm login` first before installing.
```
Comment (HonkingGoose) [inline docs/usage/private-modules.md:69]: This sentence is overly long and hard to understand.
Can we simplify this?

**Comment by HonkingGoose** [comment]
This PR is getting a lot of conflicts between `master` and my branch, __and__ I'm not sure we want to keep the npm based things in here anyways...
Do we care about any of my changes here? Or do we want to start over with the idea to generalize this away from npm?

**Comment by HonkingGoose** [comment]
@rarkins Do we want these changes still, and if so, do we want them _now_ or after you've merged `v25` into the `master` branch?

Also I think we wanted to generalize this away from `npm` specific things into a more general thing?

From issue #9205:

> Can you assess how feasible it would be to make the current private modules doc generic instead of npm-specific, or if we should create a new doc which is generic?

I'm asking because this PR is getting a lot of conflicts, and it will get conflicted again once `v25` lands into `master`. 😄

**Comment by rarkins** [comment]
@HonkingGoose I think put a pause on this until after we merge v25, then reassess which changes to keep applying. Thanks

**Comment by HonkingGoose** [comment]
I've force pushed to current `main` and basically started over, while keeping some changes from the previous commits but dropping others.

I think this PR is ready for a initial "sanity check" review. I don't know if all the text is up to date with the current behavior in the `v25` branch.

**Review by viceice — COMMENTED** [review]
I think we should make the initial description more generic and add some manager and registry specific samples at bottom.

**Comment by HonkingGoose** [comment]
We also discussed this a bit in #9205 (docs: explain better how to lookup private Maven repositories), where @rarkins said:

> Can you assess how feasible it would be to make the current private modules doc generic instead of npm-specific, or if we should create a new doc which is generic?

What do you like best @rarkins?

1. Keep npm private module docs, create new generic doc for other private modules.
2. Rework current npm private module doc to be more generic like @viceice is suggesting.

**Comment by rarkins** [comment]
I looked at this today and was thinking it's easiest to rename this to be npm specific and then start over with a new generic doc I'd write

**Comment by HonkingGoose** [comment]
I've renamed the file so that it's npm specific. This PR is ready for a review as a npm specific document now.

---

I think it makes sense for @rarkins to create a new PR _after_ this one gets merged, so that we can decide what parts we want to keep in the npm docs, and what we want to move/expand on in the generic docs.

We could copy/paste the `Self-hosted hostRules` section into that generic doc, and link to it from the npm docs, for example. That way we're not duplicating ourselves. 😄

**Comment by viceice** [comment]
You should add a new document, otherwise we have broken docs links

**Comment by rarkins** [comment]
> You should add a new document, otherwise we have broken docs links

Done. I'll try to fill this out with high priority after the merge

**Comment by rarkins** [comment]
somehow my commit is not showing up here..

**Pull request merged**